### PR TITLE
Implement reusable Claude code review workflow

### PR DESCRIPTION
**Pull request by theMickster**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-26935

## 📔 Objective

Establish Claude AI-assisted code review infrastructure for the Bitwarden Android project, including automated workflows, comprehensive development guidelines, and team ownership configuration.

### Infrastructure & Automation

- GitHub Actions workflows (.github/workflows/review-code.yml, .github/workflows/respond.yml) - Automated Claude code review triggers and response handling
  - CODEOWNERS - Routes .claude/ changes to @bitwarden/team-ai-sme

### Claude Configuration

- .claude/CLAUDE.md - Comprehensive guidelines covering:
  - Core directives (architecture adherence, code style, security requirements)
  - Module organization (:core, :data, :network, :ui, :app, :authenticator)
  - MVVM + UDF, Hilt DI, and testing patterns
  - Workflow practices and anti-patterns
- .claude/skills/reviewing-changes/SKILL.md - Structured code review skill with step-by-step guidance for compliance checks, documentation validation, and feedback generation
- .claude/prompts/review-code.md - Slash command prompt for initiating code reviews

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .claude/CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Bitwarden Android
```
Comment (theMickster) [inline .claude/CLAUDE.md]: 👋🏼 Hi @bitwarden/dept-development-mobile,
Please take some time in the coming week or so and append some detailed descriptions of our code structure, rules-of-engagement, and other key things that we want to have Claude Code leverage when it helps us with code reviews. 

Please remember to not let perfect be the enemy of good; we're aiming for short bursts of iterative success with Claude code reviews and need to think of this as a living document that we refactor as we go to ensure we get the best results from Claude. 

I'm happy to _own_ the PR, organize everyone's suggestions and input, but the guy whose been here 8 weeks cannot write this for you 😁

Comment (SaintPatrck) [reply .claude/CLAUDE.md]: I can own this. Thanks for getting it started, @theMickster

**Review by withinfocus — DISMISSED** [review]

File: .claude/skills/reviewing-changes/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   - API compatibility and deprecation warnings
+   - Resource references and manifest requirements
+
+4. **Testing** (CLAUDE.md:75): Verify appropriate test coverage
+   - Unit tests for business logic and utility functions
+   - Integration tests for complex workflows
+   - UI tests for user-facing features when applicable
+   - Test coverage for edge cases and error scenarios
+
+5. **Security** (CLAUDE.md:49-56): Given Bitwarden's security-focused nature
```
Comment (david-livefront) [inline .claude/skills/reviewing-changes/SKILL.md]: This line number is not correct

Comment (SaintPatrck) [reply .claude/skills/reviewing-changes/SKILL.md]: Good catch. Removed line numbers entirely.

File: .claude/skills/reviewing-changes/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+### Step 1: Understand Context
+
+Start with high-level assessment of the change's purpose and approach. Read PR/commit descriptions and understand what problem is being solved.
+
+### Step 2: Verify Compliance
+
+Systematically check each area against Bitwarden standards documented in CLAUDE.md:
+
+1. **Architecture**: Follow patterns in docs/ARCHITECTURE.md
```
Comment (david-livefront) [inline .claude/skills/reviewing-changes/SKILL.md]: Should `docs/ARCHITECTURE.md` be wrapped in ticks?

File: .claude/skills/reviewing-changes/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+   - MVVM + UDF (ViewModels with StateFlow, Compose UI)
+   - Hilt DI (interface injection, @HiltViewModel)
+   - Repository pattern and proper data flow
+
+2. **Style**: Adhere to docs/STYLE_AND_BEST_PRACTICES.md
```
Comment (david-livefront) [inline .claude/skills/reviewing-changes/SKILL.md]: Ditto

File: .claude/prompts/review-code.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+Use the `reviewing-changes` skill to review this pull request.
```
Comment (theMickster) [inline .claude/prompts/review-code.md:1]: @SaintPatrck 
I am liking what you done here with the skill addition. I don't know for sure if Claude is smart enough to grab that file without a direct link to it.🤷🏼‍♂️ I will keep my eye on the logs that accompany the first few PRs to ensure that this works without a link 😁 Looking great though!

Comment (SaintPatrck) [reply .claude/prompts/review-code.md:1]: As long as the SKILL.md file is in the correct place with the expected frontmatter Claude will invoke it when needed. It uses the `description` property in the frontmatter to know when and how to use the skill.

For more info about Agent skills:
- https://docs.claude.com/en/docs/claude-code/skills#what-are-agent-skills%3F
- https://docs.claude.com/en/docs/agents-and-tools/agent-skills/overview
- https://docs.claude.com/en/docs/agents-and-tools/agent-skills/quickstart
- https://docs.claude.com/en/docs/agents-and-tools/agent-skills/best-practices

Comment (theMickster) [reply .claude/prompts/review-code.md:1]: All the new-ness! Awesome!

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -10,6 +10,9 @@
 # Actions and workflow changes.
 .github/ @bitwarden/dept-development-mobile
 
+# Claude related files
```
Comment (withinfocus) [inline .github/CODEOWNERS:13]: @theMickster you've been expanding this in other PRs to include the workflows.

Comment (theMickster) [reply .github/CODEOWNERS:13]: I'll expand on what Patrick did for us here :-)

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -10,6 +10,11 @@
...
+.claude/ @bitwarden/team-ai-sme
+.github/workflows/respond.yml @bitwarden/team-ai-sme
+.github/workflows/review-code.yml @bitwarden/team-ai-sme
```
Comment (SaintPatrck) [inline .github/CODEOWNERS:16]: 👍

File: .github/CODEOWNERS (hunk 6 above)
Comment (theMickster) [reply .github/CODEOWNERS:13]: Done
